File: app/src/main/java/com/processdumper/model/ProcessInfo.java
package com.processdumper.model;

import android.content.Context;
import android.content.pm.ApplicationInfo;

import com.processdumper.utils.LogManager;

import java.util.List;

public class ProcessInfo {
    public String appName;
    public String packageName;
    public List<Integer> pids;

    public ProcessInfo(Context context, String packageName, List<Integer> pids){
        ApplicationInfo applicationInfo = null;
        try {
            applicationInfo = context.getPackageManager().getApplicationInfo(packageName,0);

        } catch (Exception e) {

        }
        finally {
            this.packageName = packageName;
            this.pids = pids;

            this.appName = (applicationInfo != null ? context.getPackageManager().getApplicationLabel(applicationInfo).toString() : "App Name not found");
        }
    }

}

File: app/src/main/java/com/processdumper/model/ProcessInfo.kt
package com.processdumper.model

import android.content.Context
import android.content.pm.ApplicationInfo
import android.content.pm.PackageManager

class ProcessInfo(){

    var AppName: String = ""
    var PackageName: String = ""

    var Pids: MutableList<Int> = mutableListOf()



    constructor(context: Context, packageName: String, pids: MutableList<Int>) : this(){
        var applicationInfo : ApplicationInfo? = null;
        try {

            applicationInfo = context.getPackageManager().getApplicationInfo(packageName, 0);

        } catch (e: PackageManager.NameNotFoundException ) {
            e.printStackTrace();
        }
        finally
        {
            PackageName = packageName;

            Pids = pids;

            if(applicationInfo != null){
                AppName = context.getPackageManager().getApplicationLabel(applicationInfo).toString();
            }
            else {
                AppName = "App Name not found";
            }
        }
    }
}